components/org.apache.stratos.autoscaler/src/main/java/org/apache/stratos/autoscaler/internal/AutoscalerServiceComponent.java

```diff
@@ -230,7 +230,7 @@ private void executeCoordinatorTasks() throws InvalidPolicyException,
             componentStartUpSynchronizer.addEventListener(new ComponentActivationEventListener() {
                 @Override
                 public void activated(Component component) {
-                    if (component == Component.StratosManager) {
+                    if (component == Component.Autoscaler) {
```

Comment:
I believe we had this waiting logic in CC and AS until SM is activated to schedule sync tasks since SM was waiting for complete topology and complete application model. With this optimization that is not needed anymore. Or am I missing something?